src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryPdao.java

```diff
@@ -252,4 +251,154 @@ public static List<BigQueryDataResultModel> aggregateTableData(TableResult resul
             });
     return values;
   }
+
+  // COLUMN STATS
+  public static final String ARRAY_TEXT_COLUMN_STATS_TEMPLATE =
+      """
+          WITH array_field AS (Select <column> FROM <table> <whereClause>)
+            Select flattened_array_field AS <column>, COUNT(*) AS <countColumn> FROM array_field CROSS JOIN UNNEST(array_field.<column>)
+            AS flattened_array_field GROUP BY flattened_array_field ORDER BY flattened_array_field <direction>
+          """;
+  public static final String TEXT_COLUMN_STATS_TEMPLATE =
+      """
+        SELECT <column>, COUNT(*) AS <countColumn> FROM <table> AS <tableName> <whereClause> GROUP BY <tableName>.<column> ORDER BY <column> <direction>
+      """;
+
+  public static final String ARRAY_NUMERIC_COLUMN_STATS_TEMPLATE =
+      """
+          WITH array_field AS (Select <column> FROM <table> <whereClause>)
+            Select MIN(flattened_array_field) AS min, MAX(flattened_array_field) AS max FROM array_field CROSS JOIN UNNEST(array_field.<column>)
+            AS flattened_array_field
+          """;
+
+  public static final String NUMERIC_COLUMN_STATS_TEMPLATE =
+      """
+        SELECT MIN(<column>) AS min, MAX(<column>) AS max FROM <table> <whereClause>
+      """;
+
+  public static ColumnStatisticsTextModel getStatsForTextColumn(
+      FSContainerInterface tdrResource,
+      String bqFormattedTableName,
+      String tableName,
+      Column column,
+      String filter)
+      throws InterruptedException {
+    String whereClause = StringUtils.isNotEmpty(filter) ? filter : "";
+    final BigQueryProject bigQueryProject = BigQueryProject.from(tdrResource);
+    final String datasetProjectId = bigQueryProject.getProjectId();
+    // The bigquery sql table name must be enclosed in backticks
+    String bigQueryTable = "`" + datasetProjectId + "." + bqFormattedTableName + "`";
```

Comment:
What do you think about delegating this formatting to the templates instead, and passing `datasetProjectId` and `bqFormattedTableName` to them?  Similar comment throughout.

Reply:
I reworked this so that we're formatting the table name in one place and sharing it with snapshots & datasets... Let's hope I didn't break everything 😁 I've added unit tests for this change.